tests/engine/test_base_game.py

```diff
@@ -158,3 +158,78 @@ def test_load_active_maps_sets_game_object_sprites():
     assert base_sheet[0][1].get_sprite().image is sprite_mock
     assert base_sheet[1][0].get_sprite().image is sprite_mock
     assert base_sheet[1][1].get_sprite().image is sprite_mock
+
+
+def test_register_player_controlled_object_correctly_registers():
+    pco = PlayerControlledObject("sprite.txt")
+
+    game = BaseGame(initial_map=generate_valid_map())
+
+    game.register_player_controlled_object(pco, 1, 2)
+
+    assert game.player_controlled_objects[pco] == (1, 2)
+    assert len(game.player_controlled_objects) == 1
+
+
+@patch("pygame.sprite.Sprite", Mock())
+def test_change_map_loads_new_player_controlled_objects_and_unloads_previous_ones():
+
+    pco1 = PlayerControlledObject("sprite.png")
+    pco2 = PlayerControlledObject("sprite.png")
+    pco3 = PlayerControlledObject("sprite.png")
+    pco4 = PlayerControlledObject("sprite.png")
+
+    go_mock = Mock()
+    go_mock.sprite_location = "sprite.png"
+
+    test_map = Map(
+        [[pco1, pco2]],
+        [[go_mock, go_mock]],
+        [[go_mock, go_mock]],
+        [[go_mock, go_mock]],
+        validate=False,
+    )
+    alt_map = Map(
+        [[pco3, pco4]],
+        [[go_mock, go_mock]],
+        [[go_mock, go_mock]],
+        [[go_mock, go_mock]],
+        validate=False,
+    )
+
+    game = BaseGame(initial_map=test_map)
+    game.object_images["sprite.png"] = Mock()  # Setup a mock to load the image as.
+    game.register_map("alt", alt_map)
+
+    game.change_map("alt")
+
+    assert pco1 not in game.player_controlled_objects
+    assert pco2 not in game.player_controlled_objects
+    assert game.player_controlled_objects[pco3] == (0, 0)
+    assert game.player_controlled_objects[pco4] == (1, 0)
+    assert len(game.player_controlled_objects) == 2
+
+
+def test_clear_player_controlled_objects_clears_them():
+    game = BaseGame(initial_map=generate_valid_map())
+    game.player_controlled_objects[PlayerControlledObject("sprite.png")] = (0, 0)
```

Comment:
Should we store the map, and register the pcos in the map for a better test?

Reply:
A pco is registered with the game (as the engine handles finding all of a Map's pcos and then registers them with the game.) 

As of https://github.com/Ross-Alexandra/TheGame/pull/20/commits/cfa221a799f2e2002c20a706e8d66fd5997d89e5 we use the ```game.register_player_controlled_object()``` method rather than manually updating the dictionary. This is better as it allows an updated way of dealing with pcos, as long as the ```register``` and ```clear``` methods are used.